kehadiran_scraper/main.py

```diff
@@ -1,93 +1,158 @@
 import asyncio
 import datetime as dt
 import logging
-import os
+from os import environ
 import typing as t
 
 import aiofiles
 import aiohttp
 
-# TODO: Read from env var for some of these
-SCRAPE_CHUNK_SIZE = 50
-SLEEP_S = 1
-FIRST_DATE = dt.date(1959, 9, 11)
+SCRAPE_CHUNK_SIZE = int(environ.get("SCRAPE_CHUNK_SIZE", 50))
+SLEEP_S = float(environ.get("SLEEP_S", 0.5))
+DISCONNECT_BACKOFF_S = float(environ.get("DISCONNECT_BACKOFF_S", 10))
+MIN_DATE = dt.datetime.strptime(environ.get("MIN_DATE", "1959-09-11"), "%Y-%m-%d").date()
+MAX_DATE = dt.datetime.strptime(environ.get("MAX_DATE", dt.datetime.today()), "%Y-%m-%d").date()
+USE_WHITELIST = bool(int(environ.get("USE_WHITELIST", "0")))
 URL_PATTERN = "https://www.parlimen.gov.my/files/hindex/pdf/DR-{day}{month}{year}.pdf"
-TEST_OUTPUT_DIR = ".test.out/"
+OUTPUT_DIR = "output/"
+WHITE_LIST_FILENAME = "whitelist-dates"
+LOG_LEVEL = environ.get("LOG_LEVEL", "DEBUG")
 
 T = t.TypeVar("T")
 
-logger = logging.getLogger(__name__)
+logger = logging.getLogger("kehadiran_scraper.main")
 
 
-def _chunker(it: t.Iterator[T], size: int) -> t.Iterator[tuple[T]]:
+def _chunker(it: t.Iterator[T], size: int) -> t.Iterator[t.Iterator[T]]:
     chunk = []
     for item in it:
         chunk.append(item)
         if len(chunk) == size:
-            yield tuple(chunk)
+            yield iter(chunk)
             chunk = []
     if chunk:
-        yield tuple(chunk)
+        yield iter(chunk)
 
 
-def _url_generator(start: dt.date) -> t.Iterator[str]:
-    today = dt.date.today()
-    days = (today - start).days
+def _url_generator(
+    min_date: dt.date,
+    max_date: dt.date,
+    whitelist: set[dt.date],
+    use_whitelist: bool,
+) -> t.Iterator[tuple[str, dt.date]]:
+    days = (max_date - min_date).days
```

Comment:
This will iterate the date in reverse right? I think would be better to iterate from oldest to now?